dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml

```diff
@@ -0,0 +1,118 @@
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTY, 7),IOC_TYPE_IOC, 0, 2)>,
+        <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTY, 6),IOC_TYPE_IOC, 0, 2)>;
+      };
+    };
+    pinmux_gpiob: pinmux_gpiob {
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTB, 18),IOC_TYPE_IOC, 0, 0)>,
+            <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTB, 19),IOC_TYPE_IOC, 0, 0)>,
+            <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTB, 20),IOC_TYPE_IOC, 0, 0)>;
+            bias-pull-down;
+      };
+    };
+  };
+
+  A group can also specify shared pin properties common to all the specified
+  pins, such as the 'bias-pull-up' property in group 2. Here is a list of
+  supported standard pin properties:
+
+  - drive-open-drain: Open-drain drive mode.
+  - bias-disable: Disable pull-up/down (default, not required).
+  - bias-pull-up: Enable pull-up resistor.
+  - bias-pull-down: Enable pull-down resistor.
+  - input-enable: enable input.
+  - input-schmitt-enable: enable input schmitt circuit.
+
+  The macro HPMICRO_PINMUX sets the pinmux function:
+  - The first parameter is the pin number, the macro HPMICRO_PIN generates pin numbers.
+  - The second parameter enables to indicate that the pin is an analogue pin.
+  - The third parameter is the remapping number,
+    look up the datasheet to obtain the parameters.
+
+compatible: "hpmicro,hpm-pinctrl"
+
+include: base.yaml
+
+child-binding:
+  description: HPMicro pin controller pin group
+  child-binding:
+    description: |
+      HPMicro pin controller pin configuration node.
+    include:
+      - name: pincfg-node.yaml
+        property-allowlist:
+          - drive-open-drain
+          - input-enable
+          - bias-disable
+          - bias-pull-down
+          - bias-pull-up
+          - input-schmitt-enable
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          Represents pin mux settings. Use HPMICRO_PINMUX
+          With:
+          - pin: The gpio pin number (0, 1, ..., GPIO_NUM_MAX)
+          - analog: bool, select analog pin in pad
+          - alt_select: alt0-alt31
+          - ioc: IOC_TYPE_IOC IOC_TYPE_BIOC IOC_TYPE_PIOC
+      drive-strength:
+        description: |
+          Pin drive capability configuration
+          High speed pin 3.3v:
+          - r000 85.61 Ohm
+          - r001 61.2 Ohm
+          - r010 42.88 Ohm
+          - r011 35.76 Ohm
+          - r111 30.67 Ohm
+          High speed pin 1.8v:
+          - r000 84.07 Ohm
+          - r001 60.14 Ohm
+          - r010 42.15 Ohm
+          - r011 35.19 Ohm
+          - r111 30.2 Ohm
+          Mormal pin:
+          - rx00 4 ma
+          - rx01 8 ma
+          - rx11 12 ma
+        type: string
+        default: "r000"
+        enum:
+          - "r000"
+          - "r001"
+          - "r010"
+          - "r011"
+          - "r100"
+          - "r101"
+          - "r110"
+          - "r111"
+      power-source:
```

Comment:
this is a standard property, see `pincfg-node.yaml`

Reply:
The properties provided by `pincfg-node.yaml` are not very useful, I deal with the same as other manufacturers here such as `NXP`, `Raspberry Pi`, etc.